csharp/HOLMS.Platform/HOLMS.Platform/Support/ReservationTags/ReservationTagBase.cs

```diff
@@ -19,6 +20,8 @@ public abstract class ReservationTagBase {
                 return string.Join(":", new List<string> { CompStayCategory }.Concat(dp));
             } else if (this is GroupBookingTag) {
                 return string.Join(":", new List<string> { GroupBookingCategory }.Concat(dp));
+            } else if (this is OTABookingTag) {
```

Comment:
This is textbook "switched polymorphism", can you please move this to the subclasses?

If you run into trouble with `ToString()`, just override it and have it call a different method.